From sgruszka@redhat.com Wed Mar 16 09:53:12 2011
From: Stanislaw Gruszka <sgruszka@redhat.com>
To: kernel@lists.fedoraproject.org, Kyle McMartin <kmcmartin@redhat.com>
Cc: Stanislaw Gruszka <sgruszka@redhat.com>, stable@kernel.org
Subject: [PATCH 2.6.35] cfg80211: fix can_beacon_sec_chan, reenable HT40
Date: Wed, 16 Mar 2011 14:52:53 +0100
Message-Id: <1300283573-3151-1-git-send-email-sgruszka@redhat.com>

From: Mark Mentovai <mark@moxienet.com>

commit 09a02fdb919876c01e8f05960750a418b3f7fa48 upstream.

Fixes 2.6.35.9 -> 2.6.35.11 regression:
https://bugzilla.redhat.com/show_bug.cgi?id=671514

This follows wireless-testing 9236d838c920e90708570d9bbd7bb82d30a38130
("cfg80211: fix extension channel checks to initiate communication") and
fixes accidental case fall-through. Without this fix, HT40 is entirely
blocked.

Signed-off-by: Mark Mentovai <mark@moxienet.com>
Cc: stable@kernel.org
Acked-by: Luis R. Rodriguez <lrodriguez@atheros.com
Signed-off-by: John W. Linville <linville@tuxdriver.com>
---
 net/wireless/chan.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/net/wireless/chan.c b/net/wireless/chan.c
index c8d190d..17cd0c0 100644
--- a/net/wireless/chan.c
+++ b/net/wireless/chan.c
@@ -54,8 +54,10 @@ static bool can_beacon_sec_chan(struct wiphy *wiphy,
 	switch (channel_type) {
 	case NL80211_CHAN_HT40PLUS:
 		diff = 20;
+		break;
 	case NL80211_CHAN_HT40MINUS:
 		diff = -20;
+		break;
 	default:
 		return false;
 	}
-- 
1.7.1

